### DISCUSSION TITLE
Adonis/Unpoly/edge architecture

### DISCUSSION THREAD

**Discussion by azurha** [unpoly/unpoly#667] [Q&A] [answered]
Hi, quick question about the Adonis/Unpoly/edge architecture.
I'm looking to have the server process and return only what is necessary, I did something like this:

Here is my register controller.

```
export default class RegistersController {
  register({ request, view }: HttpContext) {
    if (request.header('X-Up-Target')) {
      return view.render('partials/auth/register')
    }
    return view.render('pages/auth/register')
  }

  login({ request, view }: HttpContext) {
    if (request.header('X-Up-Target')) {
      return view.render('partials/auth/login')
    }
    return view.render('pages/auth/login')
  }
}
```

Does that sound right?
partials/auth/register contains only the part to be returned

pages/auth/register is a page that might look like this:

```
@let(title = "Enregistrement")
@include('partials/header')
@include('partials/top-menu')
@include('partials/auth/register')
@include('partials/footer')
```

Same logic for 'login'

Thanks

**Comment by triskweline** [comment]
Maybe @thetutlage can provide some advice.

**Comment by thetutlage** [comment, marked as answer]
@azurha I will recommend instead using components. Components have their own state so can you render them within a bigger template or the controller and provide them the needed props.

As a side note: I personally return everything and let Unpoly pick the target fragment from the response. From my experience you will never have a slow app because you are rendering full Edge templates.
